client/internal/bootstrap/client_test.go

```diff
@@ -23,382 +24,306 @@ import (
 	akssecuretlsbootstrapv1_mocks "github.com/Azure/aks-secure-tls-bootstrap/service/pkg/gen/mocks/akssecuretlsbootstrap/v1"
 	"github.com/Azure/go-autorest/autorest/adal"
 	"github.com/Azure/go-autorest/autorest/azure"
-	. "github.com/onsi/ginkgo/v2"
-	. "github.com/onsi/gomega"
+	"github.com/stretchr/testify/assert"
 	"go.uber.org/mock/gomock"
 	"go.uber.org/zap"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
 )
 
-// TODO: refactor towards vanilla go tests
-
-var _ = Describe("Client tests", Ordered, func() {
-	const (
-		apiServerFQDN  = "controlplane.azmk8s.io"
-		kubeconfigPath = "path/to/kubeconfig"
-	)
-	var (
-		ctx                 context.Context
-		mockCtrl            *gomock.Controller
-		imdsClient          *imdsmocks.MockClient
-		kubeconfigValidator *kubeconfigmocks.MockValidator
-		serviceClient       *akssecuretlsbootstrapv1_mocks.MockSecureTLSBootstrapServiceClient
-		bootstrapClient     *Client
-		bootstrapConfig     *Config
-		logger              *zap.Logger
-
-		clusterCAFilePath string
-		credFilePath      string
-	)
-
-	BeforeAll(func() {
-		logger, _ = zap.NewDevelopment()
+type BootstrapClient struct {
+	client              *Client
+	ctx                 context.Context
+	imdsClient          *imdsmocks.MockClient
+	kubeconfigValidator *kubeconfigmocks.MockValidator
+	serviceClient       *akssecuretlsbootstrapv1_mocks.MockSecureTLSBootstrapServiceClient
+}
 
-		clusterCACertPEM, _, err := testutil.GenerateCertPEM(
-			testutil.CertTemplate{
-				CommonName:   "hcp",
-				Organization: "aks",
-				IsCA:         true,
-				Expiration:   time.Now().Add(time.Hour),
+func TestBootstrapKubeletClientCredential(t *testing.T) {
+	logger, _ := zap.NewDevelopment()
+
+	tests := []struct {
+		name                   string
+		setupMocks             func(*testing.T, *gomock.Controller, *BootstrapClient, *Config) []byte
+		expectedError          string
```

Comment:
can we just have expectedError be of type `BootstrapError`? that way we don't need a separate field to denote error type